receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -25,13 +25,16 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+var gauges map[string]*metricspb.Metric
```

Comment:
I think these do need to be fields, not global variables, to allow having multiple independent receivers registered.

Reply:
Make sense, will revise